sync-core/src/test/java/com/cloudant/sync/datastore/AttachmentTest.java

```diff
@@ -42,27 +43,34 @@ public void setAndGetAttachmentsTest() throws Exception {
         MutableDocumentRevision rev_1Mut = new MutableDocumentRevision();
         rev_1Mut.body = bodyOne;
         BasicDocumentRevision rev_1 = datastore.createDocumentFromRevision(rev_1Mut);
-        File f = TestUtils.loadFixture("fixture/"+attachmentName);
+        File f = TestUtils.loadFixture("fixture/" + attachmentName);
         Attachment att = new UnsavedFileAttachment(f, "text/plain");
         BasicDocumentRevision newRevision = null;
         try {
             MutableDocumentRevision rev1_mut = rev_1.mutableCopy();
             rev1_mut.attachments.put(attachmentName, att);
             newRevision = datastore.updateDocumentFromRevision(rev1_mut);
-        } catch (ConflictException ce){
-            Assert.fail("ConflictException thrown: "+ce);
+        } catch (ConflictException ce) {
+            Assert.fail("ConflictException thrown: " + ce);
         }
         // get attachment...
+        FileInputStream fis = null;
         try {
-            byte[] expectedSha1 = Misc.getSha1(new FileInputStream(f));
+            byte[] expectedSha1 = Misc.getSha1((fis = new FileInputStream(f)));
 
-            SavedAttachment savedAtt = (SavedAttachment) datastore.getAttachment(newRevision, attachmentName);
+            SavedAttachment savedAtt = (SavedAttachment) datastore.getAttachment(newRevision,
+                    attachmentName);
             Assert.assertArrayEquals(expectedSha1, savedAtt.key);
 
-            SavedAttachment savedAtt2 = (SavedAttachment) datastore.attachmentsForRevision(newRevision).get(0);
+            SavedAttachment savedAtt2 = (SavedAttachment) datastore.attachmentsForRevision
+                    (newRevision).get(0);
             Assert.assertArrayEquals(expectedSha1, savedAtt2.key);
         } catch (FileNotFoundException fnfe) {
-            Assert.fail("FileNotFoundException thrown "+fnfe);
+            Assert.fail("FileNotFoundException thrown " + fnfe);
+        } finally {
+            if (fis != null) {
```

Comment:
Can we use `closeQuietly` here like you did before, and then you don't need to do the null check?